### PR TITLE
DOC: Improve doc for checkpointing

### LINKED ISSUE #744: What exactly is saved in '.../checkpoints/last.ckpt' when training is done for a segmentation task?
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue summary

When my segmentation task training is done, I am left with a single file containing weights: "last.cpkt" after 120 epochs using the HeadAndNeckBase class.

I used the "[Building Models](https://github.com/microsoft/InnerEye-DeepLearning/blob/main/docs/building_models.md)" guide, which says that report.ipynb uses the model checkpoint that performed best on the validation set - but it doesn't clarify if the finally saved "last.cpkt" hold the weights of the last epoch, or of the best performing checkpoint in that training session. 

### What documentation should be provided?

Would appreciate if you can clarify exactly what is saved when segmentation model training is done, so there is no doubt if I am pointing to the correct file when testing an existing model using `python Inner/ML/runner.py --model=Prostate --no-train --local_weights_path=path_to_your_checkpoint`.

Suggestion: Maybe change the name of the finally saved model to "best.cpkt" or "final.cpkt" to avoid confusion or document clearly in the section "Training a new model" if it is a requirement to the user to define what is finally saved and how it is named.

[AB#6248](https://innereye.visualstudio.com/60ce1777-00d6-4015-82bc-488a0c00202f/_workitems/edit/6248)

### PR DESCRIPTION
**Pull request by ant0nsc**
Closes #744

### CODE REVIEW

**Review by peterhessey — APPROVED** [review]
LGTM

**Review by fepegar — APPROVED** [review]
